File: java_solutions/zigzag_conversion.java
class Solution {
    public String convert(String s, int numRows) {
        //base case
        if(numRows == 1){
            return s;
        }
        StringBuilder ans = new StringBuilder();
        for(int row=0; row < numRows; row++){
            int increment = 2 * (numRows-1);
            for(int i = row; i < s.length(); i = i+increment){
                ans.append(s.charAt(i));
                if(row > 0 && row < numRows -1 && 
                i+increment - (2*row) < s.length()){
                    ans.append(s.charAt(i+increment - (2*row)));
                }
            }
        }
        return ans.toString();
        
    }
}

File: python_solutions/zigzag_conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        #base case
        if(numRows == 1):
            return s
        
        ans = ""
        for row in range(numRows):
            increment = 2 * (numRows -1)
            for i in range(row, len(s), increment):
                ans += s[i]
                #if not the first and last row
                if(row > 0 and row < numRows-1 and 
                i + increment - (2 * row) < len(s)):
                    ans += s[i + increment - (2 * row)]

        return ans 
